File: apps/ios/OvaFlus/Models/Portfolio.swift
import Foundation

struct Portfolio: Codable {
    let id: String
    var totalValue: Double
    var totalCost: Double
    var dayChange: Double
    var dayChangePercent: Double
    var holdings: [Holding]
    var lastUpdated: Date
}

struct Holding: Codable, Identifiable {
    let id: String
    var symbol: String
    var companyName: String
    var shares: Double
    var averageCost: Double
    var currentPrice: Double
    var currentValue: Double
    var dayChange: Double
    var dayChangePercent: Double
    var totalReturn: Double
    var totalReturnPercent: Double
}

struct StockQuote: Codable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let companyName: String
    var price: Double
    var change: Double
    var changePercent: Double
    var open: Double
    var high: Double
    var low: Double
    var volume: Int
    var marketCap: Double
    var peRatio: Double?
    var week52High: Double
    var week52Low: Double
}

File: apps/web/lib/api/portfolio.ts
import { apiClient } from './client';

export interface Portfolio {
  id: string;
  name: string;
  totalValue: number;
  dayChange: number;
  dayChangePercent: number;
  holdings: Holding[];
}

export interface Holding {
  id: string;
  symbol: string;
  name: string;
  shares: number;
  avgCost: number;
  currentPrice: number;
  value: number;
  change: number;
  changePercent: number;
}

export function getPortfolios() {
  return apiClient<Portfolio[]>('/portfolios');
}

export function getPortfolio(id: string) {
  return apiClient<Portfolio>(`/portfolios/${id}`);
}

export function getPortfolioPerformance(id: string, period: string = '1M') {
  return apiClient<{ date: string; value: number }[]>(
    `/portfolios/${id}/performance?period=${period}`
  );
}
